enhancements/network/sdn-live-migration.md

```diff
@@ -0,0 +1,369 @@
+  - Multitenant Isolation
+- The following features which are supported by both OpenShift SDN and OVN
+  Kubernetes but with different design and implementation. Therefore even with
+  the same name, they have different API and configure logic between OpenShift
+  SDN and OVN Kubernetes. Also, the supported platforms and modes are different
+  between the two network providers. So users need to evaluate before conducting
+  the live migration, and have to migrate the configuration manually after the
+  migration is complete. 
+  - Egress IP
+  - Egress Firewall
+  - Egress Router
+
+### User Stories
+
+The service delivery (SD) team (managed OpenShift services ARO, OSD, ROSA) has a
+unique set of requirements around downtime, node reboots, and high degree of
+automation. Specifically, SD need a way to migrate their managed fleet in a way
+that is no more impactful to the customer's workloads than an OCP upgrade, and
+that can be done at scale, in a safe, automated way that can be made
+self-service and not require SD to negotiate maintenance windows with customers.
+The current migration solution needs to be revisited to support these
+(relatively) more stringent requirements.
+
+### Risks and Mitigations
+
+## Design Details
+
+The existing ovn-kubernetes hybrid overlay feature was developed for hybrid
+Windows/Linux clusters. Each ovn-kubernetes node manages an external-to-OVN OVS
+bridge, named br-ext, which acts as the VXLAN source and endpoint for packets
+moving between pods on the node and their cluster-external destination. The
+br-ext SDN switch acts as a transparent gateway and routes traffic towards
+windows nodes.
+
+In the SDN live migration use case, we can enhance this feature to connects the
+nodes managed by different CNI plugins. To minimize the implementation effort,
+and the code maintainability, we will try to reused the whole hybrid overlay and
+only make necessary changes to both CNI plugins.
+
+On the OVN Kubernetes side, all the cross CNI traffic shall follow the same path
+of current hybrid overlay implementation. For OVN Kubernetes, we need to do
+following enhancements:
+
+1. We need to modify OVN-K to allow overlapping between the cluster network and
+   the Hybrid overlay CIDR. So that we can reuse the cluster network in the
+   migration.
+2. We need to modify OVN-K to allow modifying the hybrid overlay CIDR on the
+   fly.
+3. We need to allow `hybrid-overlay-node` running on linux nodes, currently it
+   is designed to only run on windows nodes. 
+
+On the OpenShift SDN side, when a node is converted to OVN Kubernetes,
+it shall be almost transparent to the control-plane. But we need to introduce a
+'migration mode' for OpenShift SDN by:
+
+1. Change ingress NetworkPolicy processing to be based entirely on pod IPs
+   rather than using namespace VNIDs, since packets from ovn nodes will have
+   the VNID 0 set.
+2. To be compatible with Windows node VXLAN implementation, OVN-K hybrid overlay
```

Comment:
We could figure out a way to disable this for ovnk...

Reply:
I don't know how to make it work. Even we disable it on the OVN-K side, the egress packets still cannot have the real pod MAC. In OVN-K, for all the traffic to HO nodes, the traffic will come to the cluster_router first and then route to the local br-ext bridge. So when the packets arrive at the br-ext bridge, both the src and dst mac will be changed as we cross the node subnet.

Even if we somehow manage to set the egress MACs to the real ones, I assume it will require much more code changes than we do it on the SDN side.

BTW, here is the patch I prepared for SDN https://github.com/openshift/sdn/compare/master...pliurh:sdn:live-migration.